### PR TITLE
feat: add rw_rate_limit system catalog

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
It's not easy to know what jobs are rate limited, so add this.

I want to avoid iterating over all fragments, so I introduced a new fragment flag for FsFetch. This means `rw_rate_limit` won't show fs fragments created before this PR, and I think it's acceptable.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19466?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#19466** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19466?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19466?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#19445** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19445?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19443** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19443?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/frontend/src/catalog/system_catalog/rw_catalog/rw_rate_limit.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+use risingwave_frontend_macro::system_catalog;
+
+use super::rw_fragments::extract_fragment_type_flag;
+use crate::catalog::system_catalog::SysCatalogReaderImpl;
+use crate::error::Result;
+
+#[derive(Fields)]
+struct RwFragmentRateLimit {
+    #[primary_key]
+    fragment_id: i32,
```
Comment (xxchan) [inline src/frontend/src/catalog/system_catalog/rw_catalog/rw_rate_limit.rs:25]: Maybe in theory 1 fragment can have multiple rate-limited nodes and it's not primary key

**Comment by xxchan** [comment]
Oops. Seems system catalog is not isolated between databases 🤔

image

**Comment by yezizp2012** [comment]
> Oops. Seems this system catalog is not isolated between databases 🤔
> 
> image
> Some catalogs support this, but some don't: e.g,
> 
> * `rw_tables` is isolated
> * `rw_fragments` is not

System catalogs about metadata of the streaming job are all not isolated between databases. They are used solely for debugging.

**Comment by xxchan** [comment]
Yes, I was also a little hesitant to add isolation which might hurt debugging. 🤪

**Review by yezizp2012 — APPROVED** [review]
LGTM

File: src/meta/src/controller/streaming_job.rs [hunk 2]
```diff
@@ -1778,4 +1780,107 @@ impl CatalogController {
...
+        }
+
+        Ok(rate_limits)
+    }
+}
+
+fn bitflag_intersects(column: SimpleExpr, value: i32) -> SimpleExpr {
+    column
+        .binary(BinOper::Custom("&"), value)
+        .binary(BinOper::NotEqual, 0)
```
[the remark this answers is not in the document]
Comment (xxchan) [reply src/meta/src/controller/streaming_job.rs:1881]: I checked their doc and tested locally and it works fine. 

But seems not covered in CI.

**Comment by xxchan** [comment]
### Merge activity

* **Nov 20, 3:57 PM GMT+8**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19466).

**Pull request merged**